lib/client/client.go

```diff
@@ -1562,6 +1598,52 @@ func (c *NodeClient) RunInteractiveShell(ctx context.Context, mode types.Session
 	return nil
 }
 
+// lineLabeledWriter is an io.Writer that prepends a label to each line it writes.
+type lineLabeledWriter struct {
+	linePrefix        []byte
+	w                 io.Writer
+	shouldWritePrefix bool
+}
+
+func newLineLabeledWriter(w io.Writer, label string) io.Writer {
+	return &lineLabeledWriter{
+		linePrefix:        []byte(fmt.Sprintf("[%v] ", label)),
+		w:                 w,
+		shouldWritePrefix: true,
+	}
+}
+
+func (lw *lineLabeledWriter) Write(input []byte) (int, error) {
+	out := make([]byte, 0, len(input))
+	bytesWritten := 0
+
+	for _, b := range input {
+		if lw.shouldWritePrefix {
```

Comment:
I think you can probably simplify this code by using `bytes.Split(input, []byte("\n"))`. Then you have each "line" and you don't have to manually track where and when to put the prefix.

It would also be quite a bit more efficient because you'd just be generating subslices into your existing input slice (unlike your current approach, which does a bunch of allocations)

(You might find `bytes.Cut` to be more suitable, you'd have to play with it)